src/ruby/yast/wfm.rb

```diff
@@ -247,5 +252,29 @@ def self.run_client(client)
         return false
       end
     end
+
+    # start the Ruby debugger if "Y2DEBUGGER" or "y2debugger" environment
+    # variable is set to "1", "remote" or "manual"
+    def self.debugger_from_env
+      # do not start the debugger again for each client started, run it only
+      # once for the very first client
+      return if @debugger_started
+
+      # support both upcase and down case variants, linuxrc
+      # keeps the case and it is case insensitive for the other options
+      debug = ENV["Y2DEBUGGER"] || ENV["y2debugger"]
+      return if debug != "1" && debug != "remote" && debug != "manual"
+
+      # FIXME: the UI.TextMode call is used here just to force the UI
+      # initialization, if it is initialized inside the Debugger module the
+      # ncurses UI segfaults :-(
+      # interestengly, the Qt UI works correctly...
+      Yast.import "UI"
+      log.info "text mode: #{UI.TextMode}"
+
+      log.info "Debugger set to: #{debug}"
+      Debugger.start(remote: debug == "remote", start_client: debug != "manual")
+      @debugger_started = true
+    end
```

Comment:
well, for me this is debugger code, so I expect it in debugger file and just calling from here like `Debugger.start_if_requested`